spec/services/otp_sender_spec.rb

```diff
@@ -2,27 +2,23 @@
 
 describe UserOtpSender do
   describe '#send_otp' do
-    context 'when user is two_factor_enabled and does not have unconfirmed_mobile' do
+    context 'when user does not have unconfirmed_mobile' do
       it 'sends OTP to mobile' do
-        user = build_stubbed(:user)
+        user = build_stubbed(:user, otp_secret_key: 'lzmh6ekrnc5i6aaq')
 
-        allow(user).to receive(:two_factor_enabled?).and_return(true)
-
-        expect(SmsSenderOtpJob).to receive(:perform_later).with(user)
+        expect(SmsSenderOtpJob).to receive(:perform_later).with(user.otp_code, user.mobile)
 
         UserOtpSender.new(user).send_otp
       end
     end
 
-    context 'when user is two_factor_enabled and has an unconfirmed_mobile' do
-      it 'generates a new OTP and only sends OTP to unconfirmed_mobile' do
+    context 'when user has an unconfirmed_mobile' do
+      it 'generates a new otp_secret_key and sends OTP to unconfirmed_mobile' do
         user = build_stubbed(
           :user, unconfirmed_mobile: '5005550006', otp_secret_key: 'lzmh6ekrnc5i6aaq'
         )
 
-        allow(user).to receive(:two_factor_enabled?).and_return(true)
-
-        expect(SmsSenderOtpJob).to receive(:perform_later).with(user)
+        expect(SmsSenderOtpJob).to receive(:perform_later).with(anything, user.unconfirmed_mobile)
```

Comment:
My bad about the last comment. We're already testing that. As for testing the correct code, I see the issue. It's because the new code isn't available to the expectation since it's placed before the new code is sent. This wasn't an issue before because we were checking that the user was passed in a parameter. So, in a sense, this new code makes it harder to test. I will see if there's a better way to test this.